### PR TITLE
Unify deb/rpms packaging with package-in-docker.sh and Makefile

### PR DESCRIPTION
**Pull request by justaugustus**
- Move deb and rpm build definitions under build directory
- Rename deb/rpm build scripts
- Unify deb/rpms packaging with package-in-docker.sh
- Update Makefile to include build-debs/build-rpms targets
- Enable pushes to GCS for rpms

(for discussion)
/hold
/cc @kubernetes/release-engineering @kubernetes/sig-release-admins @kubernetes/bash-firefighters

### CODE REVIEW

**Comment by justaugustus** [comment]
Test jobs to build packages added/updated here: https://github.com/kubernetes/test-infra/pull/13343

File: Makefile [hunk 1]
```diff
@@ -12,9 +12,34 @@
...
-verify-shellcheck:
+verify-shellcheck: ## Runs shellcheck
 	./hack/verify-shellcheck.sh
+
+##@ Helpers
+
+.PHONY: help
+
+help:  ## Display this help
+	@awk 'BEGIN {FS = ":.*##"; printf "\nUsage:\n  make \033[36m<target>\033[0m\n"} /^[a-zA-Z_-]+:.*?##/ { printf "  \033[36m%-15s\033[0m %s\n", $$1, $$2 } /^##@/ { printf "\n\033[1m%s\033[0m\n", substr($$0, 5) } ' $(MAKEFILE_LIST)
```
Comment (hoegaarden) [inline Makefile]: :+1:

```suggestion
	@awk ' \
		BEGIN { \
			FS = ":.*##" ; \
			col = "\033[36m" ; \
			nocol = "\033[0m" ; \
			printf "\nUsage:\n  make %s<target>%s\n", col, nocol \
		} \
		/^[a-zA-Z_-]+:.*?##/ { \
			printf "  %s%-15s%s %s\n", col, $$1, nocol, $$2 \
		} \
		/^##@/ { \
			printf "\n%s%s%s\n", col, substr($$0, 5), nocol \
		} \
	' $(MAKEFILE_LIST)
```

Maybe that's a bit more readable? Anyway, I like that!

File: build/package-in-docker.sh [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+declare -r OUTPUT_DIR="${OUTPUT_DIR:-"${PWD}/_output/${PACKAGE_TYPE}"}"
+
+case "${PACKAGE_TYPE}" in
+"debs")
+  declare -r IMG_NAME="deb-builder:${BUILD_TAG}"
+;;
+"rpms")
+  declare -r IMG_NAME="rpm-builder:${BUILD_TAG}"
+;;
```
Comment (hoegaarden) [inline build/package-in-docker.sh]:
```suggestion
;;
*)
  echo >&2 "'${PACKAGE_TYPE}' is an invalid PACKAGE_TYPE, only 'debs' and 'rpms' supported"
  exit 1
;;
```

I think we should catch invalid `PACKAGE_TYPE`s somehow

File: build/package-in-docker.sh [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+"rpms")
+  docker run -ti --rm -v "${OUTPUT_DIR}:/root/rpmbuild/RPMS/" "${IMG_NAME}" "$@"
+  echo
+  echo "----------------------------------------"
+  echo
+  echo "rpms written to: ${OUTPUT_DIR}"
+  ls -alth "${OUTPUT_DIR}"
+  echo
+  echo "yum repodata written to: "
+  ls -alth "${OUTPUT_DIR}/*/repodata/"
```
Comment (hoegaarden) [inline build/package-in-docker.sh]: That wont expand the `*`
```suggestion
  find "$OUTPUT_DIR" -type d -name repodata -print0 | xargs -0 ls -alth
```

File: build/package-in-docker.sh [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+  declare -r IMG_NAME="deb-builder:${BUILD_TAG}"
+;;
+"rpms")
+  declare -r IMG_NAME="rpm-builder:${BUILD_TAG}"
+;;
+esac
+
+docker build -t "${IMG_NAME}" "$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )/${PACKAGE_TYPE}"
+echo "Cleaning output directory..."
+rm -rf "${OUTPUT_DIR:?}/*"
```
Comment (hoegaarden) [inline build/package-in-docker.sh]: That wont expand `*`
```suggestion
find "$OUTPUT_DIR" -maxdepth 1 -mindepth 1 -print0 | xargs -0 rm -rf --
rm -rf "${OUTPUT_DIR:?}/*"
```

**Comment by tpepper** [comment]
This should have been reviewed by Google build/release folks before merge.  Eg: @listx @sumitranr 

We have no way of knowing how this impacts the actual build/release tooling since they still control it.

**Comment by listx** [comment]
This caught me by surprise earlier this week; the deb/rpm release script we use internally now needs to be updated to match these changes. For now we will use the v0.1.3 tag of this repo until the script is updated.

**Comment by justaugustus** [comment]
> This should have been reviewed by Google build/release folks before merge. Eg: @listx @sumitranr @aleksandra-malinowska aka the @kubernetes/build-admins
> 
> We have no way of knowing how this impacts the actual build/release tooling since they still control it.

@tpepper -- this has been opened for a month and Build Admins were tagged for review as part of @kubernetes/release-engineering.

**Pull request merged**